src/UXCapture.js

```diff
@@ -0,0 +1,99 @@
+import ExpectedMark from './ExpectedMark';
+import View from './View';
+
+const NOOP = () => {};
+
+class UXCapture {
+	/**
+	 * Creates a UXCapture instance
+	 *
+	 * @constructor
+	 */
+	constructor() {
+		// Store reference to instance so that we never create more than one
+		if (!UXCapture._instance) {
+			UXCapture._onMark = NOOP;
+			UXCapture._onMeasure = NOOP;
+			UXCapture._view = null;
+			UXCapture._instance = this;
+		}
+
+		return UXCapture._instance;
+	}
+
+	/**
+	 * Sets `onMark` and `onMeasure` callbacks on UXCapture singleton
+	 *
+	 * @param {object} config
+	 */
+	create(config) {
```

Comment:
I had originally been thinking that `create`, `startView`, `updateView`, `startTransition`, and `mark` would all be `static` methods of the `UXCapture` class rather than instance methods of a singleton - I'm curious what the advantage is of using instance methods that then set static properties on the class?

Thinking more about it, I'm wondering if we actually need the `class` overhead at all - why not use functional programming to really streamline the implementation using a closure that returns the public interface? I think that would be a lot easier to read and reason about rather than having to write defensive code that tries to make sure the class interface is never abused

```js
// create a UXCapture 'interface' (just a set of functions for updating UXCapture state)
function UXCapture() {
  let currentView = null;
  let onMark, onMeasure; // set by 'create'

  // methods available after `create`
  const initializedInterface = {
    startView: (zoneConfigs) => {
      currentView = new View(...);
    },
    updateView: (zoneConfigs) => {
      currentView.update(zoneConfigs);
    },
    mark: (name, waitForNextPaint = true) => {
      ...
    },
  };

  // initial public interface _only_ provides `create`
  const public = {
    create: (config) => {
      const onMark = config.onMark || NOOP;
      const onMeasure = config.onMeasure || NOOP;
      // destroy public access to `create`, expose the initialized interface
      delete public.create;
      Object.assign(public, initializedInterface);
    }
  };

  return public;
}
// return the closure
export default UXCapture();
```

There are a few ideas in there - not all of them are necessary, but at this point I don't see a need for a formal singleton.

Reply:
So this is where a clear API spec becomes important.  The one we all reviewed together says:

```
UXCapture.create
Instantiates/re-uses a singleton UXCapture instance that manages the View lifecycle.

Each View instance will manage the lifecycle of its Zones
```

Which is why I went with a formal singleton pattern. 

In your code example above, I don't quite follow. You need `mark` and `startView` in the public interface as well. 

The more I think about this, the more I think we're overcomplicating it. I think UXCapture can just be an es6 class that gets instantiated once in `ux-capture.js` and can have some logic in a constructor to prevent it from creating more than one instance.

Reply:
Ahh, I see what's happening now with ` Object.assign(public, initializedInterface);`. I still feel like it's overly complicated :(

Reply:
yeah, the `Object.assign` thing was the part I was thinking about with 'not all of them are necessary, but at this point I don't see a need for a formal singleton'.

Formal API docs shouldn't specify implementation, just interface, so it would be good to steer our API doc away from references to UXCapture instances.

The simplest version using an immutable plain object is:

```js
let currentView = null;
let onMark, onMeasure; // set by 'create'

const public = {
  create: (config) => {
    onMark = config.onMark || NOOP;
    onMeasure = config.onMeasure || NOOP;
  },
  startView: (zoneConfigs) => {
    currentView = new View(...);
  },
  updateView: (zoneConfigs) => {
    currentView.update(zoneConfigs);
  },
  mark: (name, waitForNextPaint = true) => { ... };
}
Object.freeze(public);
export default public;
```

Reply:
I went ahead with the simplest change which was basically removing a bunch of the Singleton Design Pattern boilerplate:

1) removed constructor
2) Using instance properties `this._onMark`, etc.
3) Remove `Object.freeze()` so that create() can modify the instance properties.